lib/saml_idp.rb

```diff
@@ -69,9 +69,13 @@ def signed?
         !!xpath("//ds:Signature", ds: signature_namespace).first
       end
 
-      def valid_signature?(fingerprint)
-        signed? &&
-          signed_document.validate(fingerprint, :soft)
+      def options_have_signature(options)
```

Comment:
This seems slightly unnecessary as a separate method, or at least unnecessary as a public method